src/api/objects/LegacyObjectAPIInterceptor.js

```diff
@@ -1,15 +1,56 @@
 define([
     './object-utils',
-    './ObjectAPI'
+    './ObjectAPI',
+    './objectEventEmitter'
 ], function (
     utils,
-    ObjectAPI
+    ObjectAPI,
+    objectEventEmitter
 ) {
-    function ObjectServiceProvider(objectService, instantiate) {
+    function ObjectServiceProvider(objectService, instantiate, topic) {
         this.objectService = objectService;
         this.instantiate = instantiate;
+
+        this.topicService = topic;
+        this.generalTopic = topic('mutation');
+        this.bridgeEventBuses();
     }
 
+    /**
+     * Bridges old and new style mutation events to provide compatibility between the two APIs
+     * @private
+     */
+    ObjectServiceProvider.prototype.bridgeEventBuses = function () {
+        var generalTopicListener;
+
+        var handleMutation = function (newStyleObject) {
+            var keyString = utils.makeKeyString(newStyleObject.key);
+            var oldStyleObject = this.instantiate(utils.toOldFormat(newStyleObject), keyString);
+            var specificTopic = this.topicService("mutation:" + keyString);
+
+            // Don't trigger self
+            if (generalTopicListener){
+                generalTopicListener();
+            }
+            this.generalTopic.notify(oldStyleObject);
+            specificTopic.notify(oldStyleObject.getModel());
```

Comment:
I'm not sure that this is going to be a reliable means of synchronization. It's dependent upon domain objects having at least one listener defined against them. In the case of newly created objects, this is not true. I believe that this is because extant objects have a listener defined from the tree, but new objects do not.

We may need to add a listener to each object to synchronize its model on a mutation event. Will be difficult to lifecycle the listener though.

The other option is that we just accept that even in the ye-olde parts of the system, views are going to have to be reactive to mutation. It would also address the issue of having to $digest in angular views any changes made asynchronously in new views. It shouldn't be too painful, since most stuff hangs off of BrowseController anyway.